### PR TITLE
[DOCS] Remove redirect pages

### PR DESCRIPTION
**Pull request by debadair**
Remove the redirect pages so readers don't have to go through an additional click.

### CODE REVIEW

File: docs/reference/search/point-in-time-api.asciidoc [hunk 1]
```diff
@@ -57,7 +57,7 @@ POST /_search  <1>
 // TEST[catch:unavailable]
 
 <1> A search request with the `pit` parameter must not specify `index`, `routing`,
-and {ref}/search-request-body.html#request-body-search-preference[`preference`]
+and {ref}/search-search.html[`preference`]
```
Comment (jrodewig) [inline docs/reference/search/point-in-time-api.asciidoc:60]: Unless the content is being used in another docbook, it's better to use internal cross-references.
```suggestion
or <<search-preference,`preference`>>
```

File: docs/reference/snapshot-restore/repository-azure.asciidoc [hunk 2]
```diff
@@ -2,7 +2,7 @@
 === Azure repository
 
 You can use https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction[Azure Blob storage] as a repository for
-{ref}/modules-snapshots.html[Snapshot/Restore].
+{ref}/snapshot-restore.html[Snapshot/Restore].
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/repository-azure.asciidoc:5]: For some reason, GitHub isn't letting me leave a suggestion, but I'd switch this to an internal xref:

```
<<snapshot-restore,Snapshot/Restore>>.
```

File: docs/reference/snapshot-restore/restore-snapshot.asciidoc [hunk 3]
```diff
@@ -276,7 +276,7 @@ before you start.
 
 . If you <<back-up-config-files,backed up the cluster's configuration
 files>>, you can restore them to each node. This step is optional and requires a
-<<restart-upgrade,full cluster restart>>.
+full cluster restart.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/restore-snapshot.asciidoc:279]:
```suggestion
<<restart-cluster, full cluster restart>>.
```

File: docs/reference/sql/endpoints/translate.asciidoc [hunk 4]
```diff
@@ -53,7 +53,7 @@ Which returns:
 Which is the request that SQL will run to provide the results.
 In this case, SQL will use the <<scroll-search-results,scroll>>
 API. If the result contained an aggregation then SQL would use
-the normal <<search-request-body,search>> API.
+the normal <<search-search,search>> API.
```
Comment (jrodewig) [inline docs/reference/sql/endpoints/translate.asciidoc:56]:
```suggestion
the normal <<search-search,search API>>.
```

File: docs/reference/setup/restart-cluster.asciidoc [hunk 5]
```diff
@@ -1,7 +1,7 @@
 [[restart-cluster]]
 == Full-cluster restart and rolling restart
 
-There may be {ref}/configuring-tls.html#tls-transport[situations where you want
+There may be {ref}/security-basic-setup.html[situations where you want
 to perform a full-cluster restart] or a rolling restart. In the case of
```
Comment (jrodewig) [inline docs/reference/setup/restart-cluster.asciidoc:5]: Let's leave it for now — but this isn't great link text. As a user, I have no idea where the link is taking me. 
```suggestion
There may be <<security-basic-setup,situations where you want
to perform a full-cluster restart>> or a rolling restart. In the case of
```

**Review by jrodewig — APPROVED** [review]
This looks great overall. Thanks for the update @amyjtechwriter.

Before merging, do you mind re-titling the PR and updating the description so it's intent is a bit more clear?

File: docs/reference/search/point-in-time-api.asciidoc (hunk 1 above)
Comment (amyjtechwriter) [reply docs/reference/search/point-in-time-api.asciidoc:60]: Thank you for catching that

File: docs/reference/snapshot-restore/repository-azure.asciidoc (hunk 2 above)
Comment (amyjtechwriter) [reply docs/reference/snapshot-restore/repository-azure.asciidoc]: Edited to internal ref

**Pull request merged**